src/openlmis-table/openlmis-table-component/openlmis-table.component.js

```diff
@@ -0,0 +1,99 @@
+/*
+ * This program is part of the OpenLMIS logistics management information system platform software.
+ * Copyright © 2017 VillageReach
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *  
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org. 
+ */
+
+(function() {
+
+    'use strict';
+
+    /**
+     * @ngdoc component
+     * @name openlmis-table.component:openlmisTable
+     *
+     * @description
+     * Component responsible for displaying a table
+     *
+     * @param {TableConfig} tableConfig specifies what columns, actions, header, etc. will the table have
+     * @typedef {Object} TableConfig
+     * @property {string} caption - Caption for the table.
```

Comment:
Can you use a more descriptive name here? Just from the documentation I'm not sure what caption it is. Is it a header? Is it an error message displayed when there's no data?